src/client/components/pages/RoomSchedule/RoomSchedule.tsx

```diff
@@ -0,0 +1,108 @@
+import { LocationAPI } from 'client/api/rooms';
+import { AppMessage, MESSAGE_TYPE, MESSAGE_ACTION } from 'client/classes';
+import { MenuFlex } from 'client/components/general';
+import { VerticalSpace } from 'client/components/layout';
+import { MessageContext } from 'client/context';
+import { TERM } from 'common/constants';
+import RoomResponse from 'common/dto/room/RoomResponse.dto';
+import { Combobox, NoteText } from 'mark-one';
+import React, {
+  FunctionComponent, useCallback, useContext, useEffect, useState,
+} from 'react';
+
+/**
+ * Displays the course schedule for a particular room and semester
+ */
+const RoomSchedule: FunctionComponent = () => {
+  const currentTerm = TERM.FALL;
+  const currentAcademicYear = 2020;
+  /**
+   * Saves a complete list of rooms in local state
+   */
+  const [
+    fullRoomList,
+    setFullRoomList,
+  ] = useState<RoomResponse[]>([]);
+
+  /**
+   * Keeps track of the currently selected room
+   */
+  const [
+    currentRoom,
+    setCurrentRoom,
+  ] = useState<{id: string, displayName: string}>({
+    id: '',
+    displayName: '',
+  });
+
+  /**
+   * The current value for the message context
+   */
+  const dispatchMessage = useContext(MessageContext);
+
+  const loadRooms = useCallback(async (): Promise<void> => {
+    try {
+      const roomList = await LocationAPI.getRooms();
```

Comment:
```suggestion
      const roomList = await getRooms();
```

I'd suggest importing and using `getRooms` instead of `LocationAPI.getRooms()` here as the latter was producing a type error for me (likely due to the way things are exported/imported.

Reply:
It looks like we updated all the client APIs to export the functions in this style in[ this commit](d2dc8300e07e79e3b650fdfae809c10ac00161a3) as a result of no longer being able to stub free-standing functions in Typescript 3.9.2+: 

- https://github.com/microsoft/TypeScript/issues/38568
- https://github.com/sinonjs/sinon/issues/562

What type error are you seeing in importing with `LocationAPI`?

Reply:
I was getting `Unsafe call of an any typed value` from `@typescript-eslint/no-unsafe-call`.

Reply:
I wouldn't think this would be related to Sinon as this error was thrown in the production code. I think it has more to do with the way it's being imported here (from `client/api/rooms`) and the way it's being exported (or rather isn't being exported ) from `client/api`.

Adding `export * from './rooms';` to `src/client/api/index.ts` and then changing the `import` statement to `import { LocationApi } from 'client/api'` fixes this error.

Reply:
```diff
diff --git a/src/client/api/index.ts b/src/client/api/index.ts
index f19ee6b4..4c61a767 100644
--- a/src/client/api/index.ts
+++ b/src/client/api/index.ts
@@ -6,3 +6,4 @@ export * from './nonClassMeeting';
 export * from './meetings';
 export * from './reports';
 export * from './httpStatus';
+export * from './rooms';
diff --git a/src/client/components/pages/RoomSchedule/RoomSchedule.tsx b/src/client/components/pages/RoomSchedule/RoomSchedule.tsx
index 3f9927cb..da1f7f57 100644
--- a/src/client/components/pages/RoomSchedule/RoomSchedule.tsx
+++ b/src/client/components/pages/RoomSchedule/RoomSchedule.tsx
@@ -1,4 +1,4 @@
-import { LocationAPI } from 'client/api/rooms';
+import { LocationAPI } from 'client/api';
 import { AppMessage, MESSAGE_TYPE, MESSAGE_ACTION } from 'client/classes';
 import { MenuFlex } from 'client/components/general';
 import { VerticalSpace } from 'client/components/layout';
```

Reply:
@rmainseas Ah ok thanks for looking into this. I can make the update, but for some reason I am not seeing this error - do you think we have different versions of the linter or something like that?

Reply:
Yeah of course. Not sure, that's an interesting question and probably something we should look into....